src/content/it/fundamentals/performance/rendering/reduce-the-scope-and-complexity-of-style-calculations.md

```diff
@@ -0,0 +1,188 @@
+più semplice per il browser. Nella versione precedente per sapere ad esempio,
+che l'elemento è l'ultimo del suo tipo, il browser deve prima sapere tutto su
+tutti gli altri elementi e se ci sono elementi che verranno dopo quello che
+sarebbe l'ennesimo ultimo figlio, che è potenzialmente molto più oneroso della
+semplice corrispondenza del selettore con l'elemento perché la sua classe
+corrisponde.
+
+## Riduci il numero di elementi in stile
+
+Un'altra considerazione prestazionale, che in genere è *il fattore più
+importante per molti aggiornamenti di stile* , è il volume di lavoro che deve
+essere eseguito quando un elemento cambia.
+
+In termini generali, il peggiore costo per il calcolo dello stile elaborato
+degli elementi è il numero di elementi moltiplicato per il conteggio dei
+selettori, poiché ogni elemento deve essere almeno controllato una volta su ogni
+stile per vedere se corrisponde.
+
+Note: se si modificava una classe -- ad esempio -- l'elemento body, tutti i
+figli della pagina avrebbero bisogno di ricalcolare i loro stili elaborati. Per
+fortuna non è più così, alcuni browser mantengono invece una piccola raccolta di
+regole univoche per ogni elemento che, se modificato, fa ricalcolare gli stili
+dell'elemento. Ciò significa che un elemento può o non deve essere ricalcolato a
+seconda di dove si trova nell'albero e di cosa è stato specificamente
+modificato.
+
+Spesso i calcoli di stile possono essere indirizzati a pochi elementi piuttosto
+che invalidare la pagina nel suo complesso. Nei browser moderni questo tende ad
+essere molto meno un problema perché il browser non ha necessariamente bisogno
+di controllare tutti gli elementi potenzialmente interessati da una modifica. I
+browser meno recenti, d'altra parte, non sono necessariamente ottimizzati per
+tali compiti. Dove puoi, devi **ridurre il numero di elementi invalidati** .
+
+Note: se si utilizzano componenti Web, è importante notare che i calcoli di
+stile qui sono leggermente diversi, poiché per impostazione predefinita gli
+stili non attraversano il limite del Shadow DOM e sono associati ai singoli
+componenti anziché all'intero albero. Nel complesso, tuttavia, si applica ancora
+lo stesso concetto: gli alberi più piccoli con regole più semplici vengono
+elaborati in modo più efficiente rispetto ad alberi di grandi dimensioni o
+regole complesse.
+
+## Misura il tuo costo di ricalcolo dello stile
+
+Il modo più semplice e migliore per misurare il costo dei ricalcoli di stile è
+utilizzare la modalità Timeline di Chrome DevTools. Per iniziare, apri DevTools,
+vai alla scheda Timeline, premi record e interagisci con il tuo sito. Quando
+interrompi la registrazione vedrai qualcosa di simile all'immagine qui sotto.
+
+<img
+src="images/reduce-the-scope-and-complexity-of-style-calculations/long-running-style.jpg"
+alt="DevTools showing long-running style calculations.">
+
+La striscia in alto indica fotogrammi al secondo, e se vedi le barre che vanno
```

Comment:
La striscia in alto indica i fotogrammi al secondo, e se vedi le barre che vanno